### PR TITLE
[paho-mqttpp3] update to 1.2.0

### LINKED ISSUE #15547: [paho-mqttpp3] update to 1.2.0
**Library name:** paho-mqttpp3

**New version number:** 1.2

**Other information that may be useful (release notes, etc...):**

Current version is 1.1 (released Oct 2019); the latest 1.2 release was released in Dec 2020.

https://github.com/eclipse/paho.mqtt.cpp/releases/tag/v1.2.0

### PR DESCRIPTION
**Pull request by JonLiu1993**
Fix #15547 

Update paho-mqttpp3 to the latest version 1.2.0

Note: All features are tested sucessfully in the following triplets:
1. x86-windows
2. x64-windows
3. x64-windows-static

### CODE REVIEW

**Review by NancyLi1013 — APPROVED** [review]

File: ports/paho-mqttpp3/fix-include-path.patch [hunk 1]
```diff
@@ -1,223 +1,249 @@
-diff --git a/src/mqtt/async_client.h b/src/mqtt/async_client.h
```
Comment (ras0219-msft) [inline ports/paho-mqttpp3/fix-include-path.patch:1]: Please revert the line ending changes to this file.

**Review by ras0219-msft — CHANGES_REQUESTED** [review]

**Review by NancyLi1013 — APPROVED** [review]

**Pull request merged**